### DISCUSSION TITLE
Declaração `tendo ... como`

### DISCUSSION THREAD

**Discussion by leonelsanchesdasilva** [DesignLiquido/delegua#637] [Ideias]
# Ideia geral

Suponha [uma biblioteca de arquivos](https://github.com/DesignLiquido/delegua-arquivos):

```js
const arquivos = importar('arquivos')
tendo arquivos.abrir('meu-arquivo.txt') como a {
    a.escrever('123')
}
```

No fechamento do bloco, o arquivo é fechado automaticamente.

**Comment by samuelrvg** [comment]
@leonelsanchesdasilva A implementação seria em `delegua-node` nesse caso né?

Me parece uma boa ideia.

**Reply by leonelsanchesdasilva** [reply]
Não, na verdade seria numa biblioteca em separado, e `delegua-node` teria apenas a referência a esta biblioteca.

**Reply by samuelrvg** [reply]
@leonelsanchesdasilva Quais outras funcionalidades pode ter nesse novo projeto?, pergunto pois se for somente pra abrir arquivos e manipular eles, `delegua-node` poderia fazer isso também, sem precisar de um projeto novo.

Quais outras vantagens teríamos nesse projeto separado?

**Reply by leonelsanchesdasilva** [reply]
@samuelrvg Hoje em dia todas as bibliotecas são desenvolvidas separadamente. Se eu quiser apenas usar a biblioteca fora de Delégua, eu consigo.

**Comment by leonelsanchesdasilva** [comment]
Wiki: https://github.com/DesignLiquido/delegua/wiki/Escopos#vari%C3%A1veis-controladas-por-escopos-e-a-declara%C3%A7%C3%A3o-tendo--como
